### ISSUE TITLE
ggml_conv_2d input shape

### ISSUE THREAD

**Issue by StudyingLover** [ggerganov/ggml#954]

**Comment by bssrdf** [comment]
The comments (for some reason) use Pytorch dimension convention which is opposite to what GGML has internally. 
```a->ne[2]``` and ```b->ne[2]``` are both ```IC```, the 3rd dimension ( 2nd in Pytorch).

**Comment by StudyingLover** [comment]
Okay, do we need a PR to note this detail?



---Original---
From: ***@***.***&gt;
Date: Mon, Sep 9, 2024 01:32 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ggerganov/ggml] ggml_conv_2d input shape (Issue #954)




 
The comments (for some reason) use Pytorch dimension convention which is opposite to what GGML has internally.
 a-&gt;ne[2] and b-&gt;ne[2] are both IC, the 3rd dimension ( 2nd in Pytorch).

**Comment by JohannesGaessler** [comment]
When I recently worked on `IM2COL` I didn't look at or try to understand the comments but the dimensions of `a` and `b` for a two-dimensional convolution should be in order: width, height, number of input channels, and batch size (I'm not 100% sure about the order of the first two). In `examples/mnist-common.cpp` there are asserts that let you infer at least the externally visible tensor shapes.

**Comment by StudyingLover** [comment]
yes,I used `examples/mnist-common.cpp` to help with understanding, but I believe that having correct comments is also very important.
